server/src/main/scala/io/delta/sharing/server/DeltaSharingService.scala

```diff
@@ -262,31 +273,43 @@ class DeltaSharingService(serverConfig: ServerConfig) {
       @Param("share") share: String,
       @Param("schema") schema: String,
       @Param("table") table: String,
-      queryTableRequest: QueryTableRequest): HttpResponse = processRequest {
-    if (queryTableRequest.version.isDefined && queryTableRequest.version.get < 0) {
+      request: QueryTableRequest): HttpResponse = processRequest {
+    val numVersionParams = Seq(request.version, request.timestamp, request.startingVersion)
+      .filter(_.isDefined).size
+    if (numVersionParams >= 2) {
```

Comment:
Nit: consider making this > 1, as that matches the comments better.

Reply:
updated.